src/main/java/org/generationcp/middleware/pojos/workbench/WorkbenchUser.java

```diff
@@ -99,21 +99,28 @@
 		+ "       p.project_id = :projectId "
 		+ "    GROUP BY users.userid";
 
-	public static final String GET_ACTIVE_USER_IDS_WITH_PROGRAM_ROLE_BY_PROJECT_ID =
-		"SELECT DISTINCT users.userid "
-			+ "    FROM "
-			+ "       workbench_project p "
-			+ "           INNER JOIN "
-			+ "       crop_persons cp ON cp.crop_name = p.crop_type "
-			+ "           INNER JOIN "
-			+ "       users ON cp.personid = users.personid "
-			+ "           INNER JOIN "
-			+ "       users_roles ur ON ur.userid = users.userid "
-			+ "           INNER JOIN role r ON ur.role_id = r.id  "
-			+ "   where  (r.role_type_id =  " + RoleType.PROGRAM.getId()
-			+ " 			AND ur.crop_name = p.crop_type AND ur.workbench_project_id = p.project_id) "
-			+ "    	AND p.project_id = :projectId "
-			+ "  	AND users.ustatus = 0 ";
+	public static final String GET_ACTIVE_USER_IDS_WITH_ACCESS_TO_A_PROGRAM =
+		"select distinct u.userid " //
+			+ "from users u " //
+			+ "       inner join crop_persons cp on cp.personid = u.personid " //
+			+ "       inner join users_roles ur on ur.userid = u.userid " //
+			+ "       inner join role r on ur.role_id = r.id " //
+			+ "       left join workbench_project wp on ur.workbench_project_id = wp.project_id " //
+			+ "where ((role_type_id = " + RoleType.INSTANCE.getId() + ") " //
+			+ "  || (role_type_id = " + RoleType.CROP.getId() + " and not exists(SELECT distinct p1.project_id " //
+			+ "                                      FROM workbench_project p1 " //
+			+ "                                             INNER JOIN " //
+			+ "                                      crop_persons cp1 ON cp1.crop_name = p1.crop_type " //
+			+ "                                             INNER JOIN " //
+			+ "                                      users users1 ON cp1.personid = users1.personid " //
+			+ "                                             INNER JOIN " //
+			+ "                                      users_roles ur1 ON ur1.userid = users1.userid " //
+			+ "                                             INNER JOIN role r1 ON ur1.role_id = r1.id " //
+			+ "                                      where (r1.role_type_id = " + RoleType.PROGRAM.getId() //
+			+ "                                        AND ur1.crop_name = cp.crop_name AND ur1.workbench_project_id = p1.project_id) AND users1.userid = u.userid) "
```

Comment:
Can you leave a brief code comment here explaining the new restriction and how it only applies to the combination _crop + program_ role type?